exporter/exporterhelper/internal/base_exporter.go

```diff
@@ -93,6 +93,25 @@ func NewBaseExporter(set exporter.Settings, signal pipeline.Signal, osf ObsrepSe
 		return nil, err
 	}
 
+	if be.BatcherCfg.Enabled {
+		bs := NewBatchSender(be.BatcherCfg, be.Set, be.BatchMergeFunc, be.BatchMergeSplitfunc)
+		for _, opt := range be.BatcherOpts {
+			err = multierr.Append(err, opt(bs))
+		}
+
+		if bs.mergeFunc == nil || bs.mergeSplitFunc == nil {
+			err = multierr.Append(err, fmt.Errorf("WithRequestBatchFuncs must be provided for the batcher applied to the request-based exporters"))
+		}
+
+		// Setting these callbacks to base exporter, because if they come in from options, these are only set to batch sender.
+		be.BatchMergeFunc = bs.mergeFunc
+		be.BatchMergeSplitfunc = bs.mergeSplitFunc
```

Comment:
Confused about this, can you point to the code please?

Reply:
This is an example: https://github.com/open-telemetry/opentelemetry-collector/blob/a7d019fa71199ede23838ba12ee8a29d5c7c27c3/exporter/exporterhelper/internal/batch_sender_test.go#L37

`WithRequestBatchFunc` is set to batcher but not to `base_exporter`.

Reply:
Let me clean up this API so we don't need this work around: https://github.com/open-telemetry/opentelemetry-collector/pull/11448